back/engines/commercial/analytics/app/services/analytics/populate_dimensions_service.rb

```diff
@@ -41,13 +28,9 @@ def populate_types
         ]
 
         current_types = Analytics::DimensionType.all.as_json(only: %i[name parent])
+        return unless current_types.to_set != types.as_json.to_set
 
-        return unless current_types & types != types
-
-        Analytics::DimensionType.transaction do
-          Analytics::DimensionType.delete_all
```

Comment:
Just checking: Deleting is no longer necessary?

Reply:
Not with an `insert_all` which will skip any already existing based on unique indexes - all these tables have the correct unique indexes